src/TheThingsMessage.cpp

```diff
@@ -0,0 +1,37 @@
+#include <TheThingsMessage.h>
+
+bool TheThingsMessage::decodeGetData(getdata_t *receiveData, const byte *payload, size_t length) {
+  pb_istream_t stream = pb_istream_from_buffer(payload, length);
+  const pb_field_t *fields = receiveData->has_appdata ? api_AppData_fields : api_CustomData_fields;
+  void *dest_struct;
+  if (receiveData->has_appdata) {
+    dest_struct = &receiveData->appdata;
+  } else {
+    dest_struct = &receiveData->customdata;
+  }
+  receiveData->has_appdata = false;
+  receiveData->has_customdata = false;
+  if (!pb_decode(&stream, fields, dest_struct)) {
+    return false;
+  }
```

Comment:
See above.

What is `api_CustomData_fields`? We don't know this in the `TheThingsMessage`.